src/v/redpanda/admin_server.cc

```diff
@@ -3035,6 +3035,20 @@ void admin_server::register_cluster_routes() {
           co_return ss::json::json_return_type(
             co_await map_partition_results(std::move(res.value())));
       });
+
+    register_route_sync<publik>(
+      ss::httpd::cluster_json::get_cluster_uuid,
+      [this](ss::httpd::const_req) -> ss::json::json_return_type {
+          vlog(logger.debug, "Requested cluster UUID");
+          const std::optional<uuid_t>& cluster_uuid
+            = _controller->get_storage().local().get_cluster_uuid();
+          if (cluster_uuid) {
+              ss::httpd::cluster_json::uuid ret;
+              ret.cluster_uuid = fmt::format("{}", cluster_uuid->uuid());
+              return ss::json::json_return_type(std::move(ret));
+          }
+          return ss::json::json_return_type(ss::json::json_void());
+      });
```

Comment:
Yes it would, however asynchronousity is not needed here. The intention is expressed more clearly this way IMO, so I've made use of the `path_description::set()` overload for synchronous handler.